nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java

```diff
@@ -82,14 +82,30 @@ protected void setupArgline() {
         if (configElement != null) {
             configElement.getChild("test").setValue(this.configuration.testName);
         }
-        Logger.getGlobal().log(Level.FINE, "Running surefire with: " + this.configuration.toArgLine());
-        this.mavenProject.getProperties().setProperty("argLine",
-                "" + "-Xbootclasspath/p:" + pathToNondex + File.pathSeparator
+        String argLineToSet = "" + "-Xbootclasspath/p:" + pathToNondex + File.pathSeparator
                 + Paths.get(mavenSession.getLocalRepository().getBasedir(),
                         "edu", "illinois", annotationsModuleName, ConfigurationDefaults.VERSION,
                         annotationsModuleName + "-" + ConfigurationDefaults.VERSION + ".jar")
-                        + " " + this.originalArgLine + " " + this.configuration.toArgLine());
+                + " " + this.originalArgLine + " " + this.configuration.toArgLine();
```

Comment:
Again, consider what can happen when `this.originalArgLine` sets bootclasspath. At the very least, add a TODO